### ISSUE TITLE
kubernetes_logs source metrics not available

### ISSUE THREAD

**Issue by ShahroZafar** [vectordotdev/vector#20491]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

We dont see any metrics related to kubernetes_logs source. Placing some for example below:

[k8s_format_picker_edge_cases_total](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#k8s_format_picker_edge_cases_total)
[k8s_stream_processed_bytes_total](https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#k8s_stream_processed_bytes_total)

### Configuration

```text
sources:
    kubernetes_logs:
      type: kubernetes_logs
      glob_minimum_cooldown_ms: 5000
      ingestion_timestamp_field: "ingest_timestamp"
      include_paths_glob_patterns:
        - /var/log/pods/istio-ingress_*/**
      namespace_annotation_fields: 
        namespace_labels: ""
      node_annotation_fields:
        node_labels: ""
      pod_annotation_fields:
        container_id: ""
        container_image_id: ""
    internal_metrics:
      type: internal_metrics
  sinks:
    prom_exporter:
      type: prometheus_exporter
      flush_period_secs: 30
      inputs: 
        - internal_metrics
      address: 0.0.0.0:9090
      buffer:
        type: disk
        when_full: block
        max_size: 4000000000
    kafka:
      type: kafka
      inputs:
        - kubernetes_logs
      bootstrap_servers: brokers:9092
      topic: vector-ingress-logs
      encoding:
        codec: json
      compression: zstd
      healthcheck:
        enabled: true
      batch:
        timeout_secs: 0.2
      librdkafka_options:
        client.id: "vector"
        request.required.acks: "1"
        message.timeout.ms: "0"
        linger.ms: "100"
        batch.size: "1000000"
      message_timeout_ms: 0
      buffer:
        type: disk
        when_full: block
        max_size: 4000000000
```

### Version

3.7.1

**Comment by jszwedko** [comment]
Thanks for reporting this @ShahroZafar . Some metrics were removed, but it appears we neglected to remove them from the docs. I opened https://github.com/vectordotdev/vector/pull/20530 to correct that.

`k8s_stream_processed_bytes_total` was replaced by `component_received_bytes_total`
`k8s_format_picker_edge_cases_total` is still in there but will only be reported if actually hit (that is it won't report with the initial value of `0`; it'll only report when an edge case was hit)

Hope this helps!
